modules/caddyhttp/matchers.go

```diff
@@ -118,8 +118,11 @@ func (m MatchHost) Match(r *http.Request) bool {
 		reqHost = strings.TrimSuffix(reqHost, "]")
 	}
 
+	repl := caddy.NewReplacer()
```

Comment:
This needs to be obtained from the http.Request context; see https://sourcegraph.com/github.com/caddyserver/caddy@fcd8869f51b78f260e3c3aed53d7d87c2051ffe9/-/blob/modules/caddyhttp/matchers.go#L192 for an example.

Reply:
I did not think of also using the http.Request for replacement, but mainly because there's also this place where we need to replace:

https://github.com/caddyserver/caddy/blob/fcd8869f51b78f260e3c3aed53d7d87c2051ffe9/modules/caddyhttp/caddyhttp.go#L290-L305

But this happens without any request, so we cannot use HTTP request placeholders in the host matcher, I think. But I'm also not entirely sure if we need to, I thought about it a bit and could not come up with a use case. Maybe let's only consider this when someone presents a reasonable use case for it?

Reply:
We could do replacements there, sure -- even without an HTTP request, a replacer can still replace environment variable placeholders, which exist outside of HTTP requests. But we definitely don't need to enable automatic HTTPS for dynamic host matchers (that's what On-Demand TLS is for).

Reply:
@pascalgn This line (NewReplacer()) still needs to be replaced with `repl := r.Context().Value(caddy.ReplacerCtxKey).(caddy.Replacer)` -- we have to use the existing Replacer in order to take advantage of all the HTTP-request-specific variables. It also performs better!